terraform/main.go

```diff
@@ -13,16 +13,20 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
 package main
 
+//go:generate go run ./_gen/main.go
```

Comment:
Correct me if I'm wrong, but I'm not sure generating here makes sense. Don't the gen files need to present for the program to compile, and this generation will only trigger on runtime? The provider won't compile without the files already being generated.

Reply:
Yes, they do. I think, you're correct: in case with `go:generate` we should not commit generated files, and need to generate them on the fly before build. This is not the case here: `go generate` is used as a shortcut to run _gen/main.go. For now, I moved this command to `make gen-tfschema`. It might be useful to have separate `make gen-resources` for this files.